packages/template-retail-react-app/app/pages/checkout/confirmation.jsx

```diff
@@ -34,32 +34,35 @@ import CartItemVariantImage from '../../components/item-variant/item-image'
 import CartItemVariantName from '../../components/item-variant/item-name'
 import CartItemVariantAttributes from '../../components/item-variant/item-attributes'
 import CartItemVariantPrice from '../../components/item-variant/item-price'
+import {useCurrentCustomer} from '../../hooks/use-current-customer'
+
+const onClient = typeof window !== 'undefined'
 
 const CheckoutConfirmation = () => {
+    const {orderNo} = useParams()
     const navigate = useNavigation()
-    const basket = useBasket()
-    const customer = useCustomer()
-
-    // The order data will initially be stored as our basket when first coming to this
-    // page. We capture it in local state to use for our UI. A new basket will be automatically
-    // created so we should only reference our captured order data here.
-    const [order] = useState(basket)
-
-    const form = useForm({
-        defaultValues: {
-            email: customer?.email || order?.customerInfo?.email || '',
-            password: '',
-            firstName: customer.firstName || order?.billingAddress?.firstName,
-            lastName: customer.lastName || order?.billingAddress?.lastName
+    const {data: customer} = useCurrentCustomer()
+    const register = useAuthHelper(AuthHelpers.Register)
+    const {data: order} = useOrder(
+        {
+            parameters: {orderNo}
+        },
+        {
+            enabled: !!orderNo && onClient
         }
-    })
+    )
+    const itemIds = order?.productItems.map((item) => item.productId)
+    const {data: products} = useProducts({parameters: {ids: itemIds?.join(',')}})
```

Comment:
Is it possible to have an order with zero product items? Probably not, but this will trigger a request (which will fail) if it is.

Reply:
nope! I don't think so